source/Nevermore.IntegrationTests/Model/Brand.cs

```diff
@@ -1,16 +1,25 @@
 ﻿using Nevermore.Contracts;
-using Newtonsoft.Json;
+using TinyTypes.TypedStrings;
 
 namespace Nevermore.IntegrationTests.Model
 {
-    public abstract class Brand : IDocument
+    public abstract class Brand : IDocument<BrandId>, IId
     {
-        public string Id { get; protected set; }
+        public BrandId Id { get; protected set; }
         public string Name { get; set; }
 
         public abstract string Type { get; }
 
         public string Description { get; set; }
+
+        string IId.Id => Id?.Value;
```

Comment:
This is ugly, but pragmatic. 

Here we explicitly implement the old non-generic interfaces (`IId`/`IDocument`) as well, which require a strictly `string` ID. This allows for reuse of the old succinct Nevermore signatures without adding type parameters where they don't provide value. This choice also happens to provide for backwards compatibility.

Reply:
Good idea via @TomPeters to avoid this explicit `IId.Id` implementation (please correct me if my summary is wrong though Tom): only implement `IId`, and remove the `string Id` property and `IIdWrapper` entirely - rely on the IdColumn mapping to read/write to the `Id` property appropriately. 

I'll give this a go and see if I hit any sharp corners.

Reply:
I've pushed this change here: https://github.com/OctopusDeploy/Nevermore/pull/87/commits/35f57a50b798d711c88138cd69b9bd81827ebf96. Keen for feedback.

Reply:
Unsurprisingly, this breaks a _bunch_ of usages of `IId.Id` in `OctopusDeploy` (WIP in https://github.com/OctopusDeploy/OctopusDeploy/pull/5392).